erpnext/public/js/projects/project_tree.js

```diff
@@ -0,0 +1,1247 @@
+	}
+
+	get_task_list_row_html(doc, level) {
+		return this.get_task_list_row_html_skeleton(this.get_left_html(this.task_columns, doc, level), this.get_right_html(doc, true), doc, level);
+	}
+
+	get_list_row_html_skeleton(left = '', right = '', doc = {}) {
+		return `
+			<div class="list-row-container project-list-row-container" tabindex="1" data-doctype="Project" data-name="${escape(doc.name)}">
+				<div class="level list-row small">
+					<div class="level-left ellipsis">
+						${left}
+					</div>
+					<div class="level-right text-muted ellipsis">
+						${right}
+					</div>
+				</div>
+			</div>
+		`;
+	}
+
+	get_task_list_row_html_skeleton(left = '', right = '', doc = {}, level) {
+		return `
+		<div class="list-rows" data-doctype="Task" data-name="${escape(doc.name)}" data-level="${level}">
+			<div class="list-row-container" tabindex="1">
+				<div class="level list-row small">
+					<div class="level-left ellipsis">
+						${left}
+					</div>
+					<div class="level-right text-muted ellipsis">
+						${right}
+					</div>
+				</div>
+			</div>
+			<div class="nested-list-row-container hide">
+			</div>
+		</div>
+		`;
+	}
+
+	get_left_html(columns, doc, level) {
+		return columns.map(col => this.get_column_html(col, columns, doc, level)).join('');
+	}
+
+	get_right_html(doc, create_new=false) {
+		return this.get_meta_html(doc, create_new);
+	}
+
+	get_column_html(col, columns, doc, level) {
+		if (col.type === 'Status') {
+			return `
+				<div class="list-row-col hidden-xs ellipsis">
+					${this.get_indicator_html(doc)}
+				</div>
+			`;
+		}
+
+		const df = col.df || {};
+		const label = df.label;
+		const fieldname = df.fieldname;
+		const value = doc[fieldname] || '';
+
+		const format = () => {
+			if (df.fieldtype === 'Code') {
+				return value;
+			} else if (df.fieldtype === 'Percent') {
+				return `<div class="progress level" style="margin: 0px;">
+						<div class="progress-bar progress-bar-success" role="progressbar"
+							aria-valuenow="${value}"
+							aria-valuemin="0" aria-valuemax="100" style="width: ${Math.round(value)}%;">
+						</div>
+					</div>`;
+			} else {
+				return frappe.format(value, df, null, doc);
+			}
+		};
+
+		const field_html = () => {
+			let html;
+			let _value;
+			// listview_setting formatter
+			if (this.settings.formatters && this.settings.formatters[fieldname]) {
+				_value = this.settings.formatters[fieldname](value, df, doc);
+			} else {
+				let strip_html_required = df.fieldtype == 'Text Editor'
+					|| (df.fetch_from && ['Text', 'Small Text'].includes(df.fieldtype));
+				if (strip_html_required) {
+					_value = strip_html(value);
+				} else {
+					_value = typeof value === 'string' ? frappe.utils.escape_html(value) : value;
+				}
+			}
+
+			if (df.fieldtype === 'Image') {
+				html = df.options ?
+					`<img src="${doc[df.options]}" style="max-level: 30px; max-width: 100%;">` :
```

Comment:
@DeveloperMetal would it be ok to add inline CSS here or should this be on a separate file?

Reply:
I would prefer we do not inline any css(unless it depends on dynamic values). that's just dirty and prevents theming from just css files.